controlplane/pkg/apis/registry/registry.proto

```diff
@@ -17,6 +18,15 @@ message NetworkServiceEndpoint {
 message NetworkService {
     string name = 1;
     string payload = 2;
+    message Match {
+        k8s.io.apimachinery.pkg.apis.meta.v1.LabelSelector source_selector = 1;
+        message Destination{
```

Comment:
Could we define this not-inline in another message?  It turns out to make the generated Go super messy.